### PR TITLE
X bit support

### LINKED ISSUE #9: Directory +x / Files not +x
Hi

It does not seem possible to do `setfacl -R -m u:myuser:rwX myfolder`

Any idea (except no-recurse/give all subfiles) ? 

Best regards

### PR DESCRIPTION
**Pull request by jadestorm**
#### Pull Request (PR) description
This pull request includes a fix that @pyslarvt made to add rwX style settings that didn't appear to make it into a formal PR.  The fix as-is would frequently leave my hosts updating on every Puppet run.  The modifications I made add in support for properly comparison the current permissions set, even if you have the capital X bit set.  (note: X is only for setfacl, it's not a "real" posix acl, so you have to lowercase it to do comparisons)  This was already handled in the :unset handler, but not in set or exact.  Note it is possible that I've missed a few use cases as I don't really use anything but the set action.

I may follow this PR by writing up a quick README as well, depends on what time I have to spare.  =)

#### This Pull Request (PR) fixes the following issues
Fixes #9

### CODE REVIEW

**Comment by jadestorm** [comment]
[ps hold tight on accepting this, dealing with rubocop complaints and such -- going to condense the commits once i finish resolving it]

**Comment by jadestorm** [comment]
Ok we should be good!

File: lib/puppet/type/posix_acl.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 require 'set'
 require 'pathname'
+require 'English'
```
Comment (bastelfreak) [inline lib/puppet/type/posix_acl.rb]: why is this needed?

Comment (ekohl) [reply lib/puppet/type/posix_acl.rb]: It gives you `$LAST_MATCH_INFO`

File: lib/puppet/type/posix_acl.rb [hunk 2]
```diff
@@ -161,8 +162,10 @@ def unset_insync(cur_perm)
       (sp - cp).sort == sp
     end
 
+    # Make sure we are not misinterpreting recursive permission notation (e.g. rwX) when
+    # comparing current to new perms.
     def set_insync(cur_perm) # rubocop:disable Style/AccessorMethodName
-      should = @should.uniq.sort
+      should = @should.uniq.map(&:downcase).sort
```
Comment (ekohl) [inline lib/puppet/type/posix_acl.rb]: perhaps better to first downcase and then uniq

File: lib/puppet/type/posix_acl.rb (hunk 1 above)
Comment (jadestorm) [reply lib/puppet/type/posix_acl.rb]: And really the only reason I switched from $~ to $LAST_MATCH_INFO because rubocop yelled at me and questioned my lineage.  ^_^

File: lib/puppet/type/posix_acl.rb (hunk 2 above)
Comment (jadestorm) [reply lib/puppet/type/posix_acl.rb]: Excellent point  =D  Changed!
